manticore/native/cpu/arm.py

```diff
@@ -455,12 +464,7 @@ def _shift(cpu, value, _type, amount, carry):
             amount = 1
 
         elif _type in range(cs.arm.ARM_SFT_ASR_REG, cs.arm.ARM_SFT_RRX_REG + 1):
-            if cpu.mode == cs.CS_MODE_THUMB:
-                src = amount.read()
-            else:
-                src_reg = cpu.instruction.reg_name(amount).upper()
-                src = cpu.regfile.read(src_reg)
-            amount = Operators.EXTRACT(src, 0, 8)
+            amount = Operators.EXTRACT(amount, 0, 8)
```

Comment:
So the `amount` would be a register that stores the shift value now?

Reply:
Note that below we use  e.g. `ASR_C` from `bitwise.py` and this expcets amount to be a number; is it always fine?

Reply:
I won't have much time to look into this in the near future due to other priorities, but to address your point about `bitwise`: there are tests for ASR and ROR in the testsuite, which pass with this change. That's why the change at `manticore/native/cpu/arm.py:105` was necessary. But it's hard to say for sure whether this would work for all inputs.